solutions/src/main/java/ru/sberbank/school/task10/MyScalableThreadPool.java

```diff
@@ -0,0 +1,189 @@
+
+        IntStream.range(size, sizeMax)
+                .forEach(freeIndexForNewWorker::add);
+    }
+
+    @Override
+    public void start() {
+        poolMap.forEach((key, worker) -> {
+            worker.start();
+
+            synchronized (activeThreadsLock) {
+                activeThreads++;
+            }
+        });
+    }
+
+    @Override
+    public void stopNow() {
+        synchronized (poolMap) {
+            poolMap.forEach((key, worker) -> worker.interrupt());
+            poolMap.clear();
+            freeIndexForNewWorker.clear();
+        }
+        synchronized (tasks) {
+            tasks.clear();
+        }
+        activeThreads = 0;     //  before tasks.clear() sometimes "activeThreads = 1"
+                               // in last reading... after "= 0" writing    NB!!! repeat hb rule !!!
+    }
+
+    @Override
+    public void execute (Runnable runnable){
+        synchronized (tasks) {
+            tasks.add(new FutureTask(runnable, true));
+            tasks.notifyAll();
+        }
+    }
+
+    @Override
+    public <T> Future <T> execute(Callable <T> callable) {
+        synchronized (tasks) {
+            FutureTask<T> futureTask = new FutureTask(callable);
+            tasks.add(futureTask);
+            tasks.notifyAll();
+            return futureTask;
+        }
+    }
+
+    /*
+    *return true if needed to interrupt, otherwise return false
+    *  */
+    private boolean poolResize () {
+        int taskSize = tasks.size();
+
+        synchronized (poolMap) {
+            poolMap.forEach((key, worker) -> {
+
+                if (worker == null || worker.isInterrupted()) {
+                    worker = null;
+                    if (!freeIndexForNewWorker.contains(key)) {
+                        freeIndexForNewWorker.offer(key);
+                    }
+
+                    synchronized (activeThreadsLock) {
+                        activeThreads--;
+                    }
+                }
+            });
+
+            if (taskSize > sizeMin && activeThreads < sizeMax) {
+                synchronized (activeThreadsLock) {
+                    activeThreads++;
+                }
+
+                if (!freeIndexForNewWorker.isEmpty()) {
+                    int id = freeIndexForNewWorker.poll();
+
+                    ThreadWorker worker =new ThreadWorker("ThreadWorker " + id, id);
+                    poolMap.put(id, worker);
+                    worker.start();
+                }
+            }
+
+            return taskSize < activeThreads && activeThreads > sizeMin;
+        }
+    }
+
+    public int getActiveThreads () {
+        return activeThreads;
+    }
+
+    private class ThreadWorker extends Thread {
+        private int id;
+
+        private ThreadWorker(String name, int id) {
+            super(name);
+            this.id = id;
+        }
+
+        @Override
+        public void run() {
+            System.out.println("ThreadWorker "+ id + " started");
```

Comment:
WhitespaceAround: &#39;+&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-2031494748>*